### PR TITLE
Added metric name to GetObservationLogRequest

### PR DESCRIPTION
**Pull request by johnugeorge**
Added optional field 'metric_name' to GetObservationLogRequest in order to filter logs based on metric name if set

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/559)

### CODE REVIEW

File: pkg/api/v1alpha2/api.proto [hunk 1]
```diff
@@ -491,8 +492,9 @@ message ReportObservationLogReply {
 
 message GetObservationLogRequest {
     string trial_name = 1;
-    string start_time = 2; ///The start of the time range. RFC3339 format
-    string end_time = 3; ///The end of the time range. RFC3339 format
+    string metric_name = 2;
```
Comment (hougangliu) [inline pkg/api/v1alpha2/api.proto]: how about `s/string metric_name/repeat string metric_name_list` to make it more scalable

Comment (johnugeorge) [reply pkg/api/v1alpha2/api.proto]: Can we target this enhancement in a separate PR? Currently, controller will need to pass only one metric name which would be the objective metric name.

Comment (hougangliu) [reply pkg/api/v1alpha2/api.proto]: sure

**Pull request merged**